recipes/doubtlab/meta.yaml

```diff
@@ -0,0 +1,114 @@
+{% set name = "doubtlab" %}
+{% set version = "0.1.1" %}
+{% set sha256 = "e2065c916820ac6008883032594c14a0a660862cb012f3fac152a97f4c3c64b4" %}
+
+##----------------------------------------------------------------------##
+## TODO for Maintainers:
+##
+## >> After each PyPI release of the package (doublab):
+## 
+##    1. Get the corresponding "version" and "sha256" of the latest 
+##       release and update the jinja2 variables (line number: 2 and 3).
+##       - version
+##       - sha256
+##   
+##       Note: run the following commands from within the recipes/doubtlab. 
+##             $ . ./fetch_sha256.sh
+##             $ pypi_sha256 "doubtlab" "0.1.1" # this will give you the sha256  
+##
+##    2. Crosscheck and update the package dependency versions under:
+##       - requirements.host
+##       - requirements.run
+##
+## SEE THIS EXAMPLE: 
+## - how "pysteps" is maintaining "conda-forge/pysteps-feedstock":
+## https://pysteps.readthedocs.io/en/latest/developer_guide/update_conda_forge.html
+##
+##----------------------------------------------------------------------##
```

Comment:
Conda-forge has a bot that will automatically create a PR for a new release every time one is pushed to PyPI, you won't need to update the `sha256` and version yourself. The only thing you'll need to pay attention to is the requirements and license (If you change that).

Reply:
Example: https://github.com/conda-forge/lark-feedstock/pull/2

Reply:
@BastianZim Thank you, for clarifying that and reviewing. Should I remove the `fetch_sha256.sh` then, or is it okay to leave it there? What do you suggest?

Also, (assuming you meant the `@auto-tick` bot) is there some script that could extract the dependency-versions directly from the package and update the requirements?

Something that I am not clear about is, given I have some shell script to extract the requirements from the PyPI package itself, how can I update that in the `meta.yaml` file?

Reply:
I would remove it and let the bot handle it.

Unfortunately not. You can use Grayskull for this but this is something you'll need to manually double-check. The bot-PR will also include a check but it's experimental and not guaranteed to work 100%.

Basically, a version update works like this:  

1. A new version is published on PyPI.
2. The conda-forge bot picks this up (Note, this can take a couple of hours sometimes. The status page for this is https://conda-forge.org/status/#version_updates
3. The bot opens a new PR, updating the version and `sha256`.
4. The maintainer verifies that no requirements changed and that the license and, if provided, URLs are still the same.
5. The maintainer merges the PR.

I'm not sure what you mean by updating the meta.yaml file. This is done via a normal PR (Please use a personal fork for this, as each push is uploaded to Anaconda.org, no matter which branch).

Reply:
Thank you. I removed the `fetch_sha256.sh` file.

Reply:
Thank you, for explaining the version-update process-flow.